sycl/include/sycl/ext/intel/math.hpp

```diff
@@ -61,6 +79,138 @@ std::enable_if_t<std::is_same_v<Tp, sycl::half>, sycl::half> copysign(Tp x,
   return __builtin_bit_cast(sycl::half, __imf_copysignf16(xi, yi));
 }
 
+template <typename Tp>
+std::enable_if_t<std::is_same_v<Tp, float>, float> ceil(Tp x) {
+  return __imf_ceilf(x);
+}
+
+template <typename Tp>
+std::enable_if_t<std::is_same_v<Tp, double>, double> ceil(Tp x) {
+  return __imf_ceil(x);
+}
+
+template <typename Tp>
+std::enable_if_t<std::is_same_v<Tp, sycl::half>, sycl::half> ceil(Tp x) {
+  static_assert(sizeof(sycl::half) == sizeof(_iml_half_internal),
```

Comment:
Is it possible to avoid the multiplication of this line in all functions? It looks the same for all instances.
Just to make this header file shorter and more clear.

Reply:
Hi, @akolesov-intel 
I moved static_assert check to sycl::ext::intel::math namespace, the check in all half/half2 functions are removed now.
Thanks very much.

Reply:
Thanks a lot, @jinge90 !